src/handlers.rs

```
@@ -107,25 +125,30 @@ impl FromStr for ServiceWithName {
 
 impl ServiceWithName {
     fn lookup(&self) -> Result<Option<Service>> {
-        let serv_entry: *const servent;
+        let service_cstr = CString::new(self.service.as_str())?;
+        let proto_cstr = self.proto.as_ref().map(|s| CString::new(s.as_str()).unwrap());
```

`unwrap()` can panic, we should return an error instead